### PR TITLE
Issues 1239/move to travis

### PR DESCRIPTION
**Pull request by dlebrero**
Moving build to Travis.

The build steps are now:

1. Run the Backend unit tests
2. Build the Backend production image
3. Run the Client unit tests + linting 
4. Build the Client production image
5. Build the Windshaft production image
6. Start a Docker Compose environment for additional testing
7. Run the Backend functional tests (using the previous env)
8. Run the end to end tests (using the previous env)
9. If master or develop branch, publish containers and deploy

The most significant change is that now the end to end tests run against the production images. This means:
1.  We test the production images before using them in production
2. We avoid the intermittent issue with the default Postgres DB that comes with Travis/CircleCI
3. The build takes 10 minutes instead of 6
4. The whole build can be tested and run locally
5. Some minor changes in the prod code, for example to not require Sentry DSN

We have also changed the way we build the production images. We now use the developer image to create the assets that will be use in the production images. This means:
1. Building the production images just requires docker, no need to install lein/jvm
2. Removed the duplication between the client dev and prod images
3. The Client image is significantly smaller: from 600mb to 50mb

### CODE REVIEW

File: backend/import-and-run.sh [hunk 1]
```diff
@@ -2,4 +2,13 @@
...
 keytool -import -trustcacerts -keystore /usr/lib/jvm/java-1.8-openjdk/jre/lib/security/cacerts -storepass changeit -noprompt -alias postgrescert -file /pg-certs/server.crt
 
-./run-as-user.sh lein repl :headless
+if [ -z "$1" ]; then
+    ./run-as-user.sh lein repl :headless
+elif [ "$1" == "functional-and-seed" ]; then
+    # Two thing in one so that we avoid starting yet another JVM
+    ./run-as-user.sh lein do test :functional, run -m dev/migrate-and-seed
+elif [ "$1" == "kubernetes-test" ]; then
+    ./run-as-user.sh lein test :kubernetes-test
```
Comment (iperdomo) [inline backend/import-and-run.sh]: Do we have tests annotated with `:kubernetes-test`?

File: client/ci-build.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
```
Comment (iperdomo) [inline client/ci-build.sh]: if you use `#!/usr/bin/env bash` like the other script you can use the keyword `function`. I guess we should standardize to use bash, right?

File: backend/import-and-run.sh (hunk 1 above)
Comment (dlebrero) [reply backend/import-and-run.sh]: No, I was planning to add one but now I realize that those will be with puppeteer.

File: client/ci-build.sh (hunk 2 above)
Comment (dlebrero) [reply client/ci-build.sh]: Bash was/is not available in some of the prod containers

Comment (dlebrero) [reply client/ci-build.sh]: Ignore the previous comment

**Pull request merged**